phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
```
$result = $ext->get_errrors();

if ($result === true) {
     // continue processing
} else {
    $errs = [];
    if (is_string($result) || is_array($result)) {
        foreach ((array) $result as $message) {
             if (is_array($message)) {
                 $errs[] = $language->lang_array($message[0]; $message[1]);
             } else {
                 $errs[] = $language->lang($message);
             }
        }
    } else {
        $errs[] = $language->lang('GENERIC_ERROR_MESSAGE');
    }
}
```

It is a bit complex because it has to deal with a lot of different cases but you can simplify it if you force the most complex form for the errors and assume you will never have anything else (may be a bad idea) :

```
$result = $ext->get_errrors();

if ($result === true) {
     // continue processing
} else {
    $errs = [];
        foreach ((array) $result as $message) {
             $errs[] = $language->lang_array($message[0]; $message[1]);
        }
}
```

For your other concern, a string does not mean true, not in the real world and it does not mean false either. It is matter of convention and having true for a success or a list of errors in case of error is a common convention in software design.

But as I also said, it's not really the best convention. This kind of error could be solve another way:
 - pre_enable() / pre_disable() / pre_purge() methods
 - these methods are not expected to return anything
 - these methods do nothing by default
 - these methods can throw an exception to stop the installer from going forward 
 - the exception can implement the phpbb exception_interface, and in that case the language var will be handled
 - if we really want to support multiple error message we can add a new class of exception to deal with it